common/lib/protocol-definitions/src/clients.ts

```diff
@@ -3,66 +3,181 @@
  * Licensed under the MIT License.
  */
 
+import { ScopeType } from "./scopes";
 import { IUser } from "./users";
 
+/**
+ * A client's connection mode - either view-only ("read") or allowing edits ("write").
+ *
+ * @remarks
+ *
+ * Note: a user's connection mode is dependent on their permissions.
+ * E.g. a user with read-only permissions will not be allowed a "write" connection mode.
+ */
 export type ConnectionMode = "write" | "read";
 
+/**
+ * Capabilities of a Client.
+ * In particular, whether or not the client is {@link ICapabilities.interactive}.
+ */
 export interface ICapabilities {
+	/**
+	 * Indicates if the client represents a potentially interactive session with a user (if 'true') or if it's
+	 * a "system entity" (if 'false').
+	 *
+	 * @remarks
+	 *
+	 * The only "system entity" scenario at the moment is the
+	 * {@link https://fluidframework.com/docs/concepts/summarizer/ | summarizer client}.
+	 */
 	interactive: boolean;
 }
+
+/**
+ * The kind of client connected to the service.
+ */
+export type ClientType = "client" | "summarizer";
+
+/**
+ * {@link IClient} connection / environment metadata.
+ */
 export interface IClientDetails {
+	/**
+	 * {@inheritDoc ICapabilities}
+	 */
 	capabilities: ICapabilities;
-	type?: string;
+
+	/**
+	 * The kind of client being described.
+	 *
+	 * `undefined` indicates that the kind could not be determined.
+	 *
+	 * @remarks
+	 *
+	 * General `string` values are allowed for type-wise backwards compatibility, but this support
+	 * will be removed in the future.
+	 */
+	type?: ClientType | string;
+
 	/**
+	 * TODO: What are the semantics of this?
+	 *
+	 * @remarks
+	 *
 	 * If the environment needs to specify multiple properties which gives info about the environment, then
-	 * it should be in particular format like: "prop1:val1;prop2:val2;prop3:val3"
+	 * it should be in particular format like: `prop1:val1;prop2:val2;prop3:val3`.
+	 *
+	 * @example TODO: an example here would be helpful, since the expected form is complex and the type is simply `string`
+	 *
+	 * Note: this is optional for backwards compatibility, but will be required in the future.
```

Comment:
It appears as though this is always provided. Do we want to require this? If not, what are the semantics of it being absent?